.github/workflows/artifacts-build.yml

```diff
@@ -195,36 +194,349 @@ jobs:
         echo "https://storage.googleapis.com/icu4x-pr-artifacts/gha/${{ github.sha }}/wasm-demo/index.html"
         echo "::endgroup::"
 
+  bench-perf:
+    # This is too expensive to run on every push, so only run it on main.
+    # Remove this line to debug.
+    if: github.ref == 'refs/heads/main' && github.repository == 'unicode-org/icu4x'
+    concurrency:
+      # Allow one run at a time to include the previous run's results
+      group: bench-perf
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        component:
+          - components/locid
+          - components/plurals
+          - components/datetime
+          - components/collections
+          - utils/fixed_decimal
+
+    # If you are modifying and debugging is required, don't be afraid to get
+    # messy in a personal fork, if no better way to do it.
+    # Example "debugging" workflow: https://github.com/echeran/icu4x/actions/runs/296714990
+
+    steps:
+    - uses: actions/checkout@v3
+
+    # Cargo-make boilerplate
+    - name: Get cargo-make version
+      id: cargo-make-version
+      run: |
+        echo "hash=$(cargo search cargo-make | grep '^cargo-make =' | md5sum)" >> $GITHUB_OUTPUT
+      shell: bash
+    - name: Attempt to load cached cargo-make
+      uses: actions/cache@v3
+      id: cargo-make-cache
+      with:
+        path: |
+          ~/.cargo/bin/cargo-make
+          ~/.cargo/bin/cargo-make.exe
+        key: ${{ runner.os }}-make-${{ steps.cargo-make-version.outputs.hash }}
+    - name: Install cargo-make
+      if: steps.cargo-make-cache.outputs.cache-hit != 'true'
+      run: cargo +stable install cargo-make
+
+    # Actual job
+
+    - name: Run benchmark
+      run: |
+        pushd $PWD && cd ${{ matrix.component }};
+        export REL_OUTPUT_PATH="`dirs +1`/benchmarks/perf/${{ matrix.component }}";
+        mkdir -p $REL_OUTPUT_PATH;
+        export OUTPUT_PATH_CMD="ls -d $REL_OUTPUT_PATH";
+        export OUTPUT_PATH=$(echo $OUTPUT_PATH_CMD | sh);
+        cargo bench --features bench -- --output-format bencher | tee $OUTPUT_PATH/output.txt;
+
+    - name:  Download previous benchmark data
+      run: |
+        mkdir -p benchmarks/perf
+        gsutil -m cp -rn gs://${{ env.GCP_BUCKET_ID }}/gha/$(git log --pretty=format:'%H' -1 -r HEAD^)/benchmarks/perf/${{ matrix.component }} benchmarks/perf
```

Comment:
Note: before merging this, we have to upload `merged-bench-data` to GCP under `main`'s current hash.